### PR TITLE
Remember User Preferences Branch

### PR DESCRIPTION
**Pull request by kevintbradbury**
#81 Remember User Preferences

### CODE REVIEW

File: BlackNectar/BlackNectar.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -351,7 +370,7 @@
 				TargetAttributes = {
 					321319391DE0F61500FB1774 = {
 						CreatedOnToolsVersion = 8.1;
-						DevelopmentTeam = 3TAADZBCK9;
+						DevelopmentTeam = 5E82HJH2N5;
```
Comment (SirWellington) [inline BlackNectar/BlackNectar.xcodeproj/project.pbxproj:420]: Once again, watch the development team. We shouldn't be changing this with each PR. We need to set it to none.

File: BlackNectar/BlackNectar/SideMenuFilterViewController.swift [hunk 2]
```diff
@@ -53,14 +54,18 @@ class SideMenuFilterViewController: UITableViewController {
     }
     
     @IBAction func sliderDidSlide(_ sender: UISlider) {
-
+        
+        UserPreferences.instance.distanceFilter.subtract(distanceFilter)
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/SideMenuFilterViewController.swift:58]: Should it subtract, or should it just set it?

Comment (SirWellington) [reply BlackNectar/BlackNectar/SideMenuFilterViewController.swift:58]: It's not exactly clear to me the reason behind the subtract & add.

File: BlackNectar/BlackNectar/SideMenuFilterViewController.swift [hunk 3]
```diff
@@ -69,11 +74,11 @@ class SideMenuFilterViewController: UITableViewController {
 
         if openNowSwitch.isOn {
 
-            isOpenNow = true
-
+            isOpenNow = true; UserPreferences.instance.isOpenNow = true
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/SideMenuFilterViewController.swift]: Second statement should be on its own line.

File: BlackNectar/BlackNectar/SideMenuFilterViewController.swift [hunk 4]
```diff
@@ -83,19 +88,19 @@ class SideMenuFilterViewController: UITableViewController {
 
         if isRestaurant == false {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/SideMenuFilterViewController.swift:91]: It's a little weird to compare boolean values to true or false. It is much more natural to just say:
```swift
if isRestaurant {
}
```

File: BlackNectar/BlackNectar/SideMenuFilterViewController.swift [hunk 5]
```diff
@@ -83,19 +88,19 @@ class SideMenuFilterViewController: UITableViewController {
...
-
+            
         } else if isRestaurant == true {
 
-            isRestaurant = false
+            isRestaurant = false; UserPreferences.instance.isRestaurant = false
             styleButtonOff(button: restaurantButton)
-
+            
         } else {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/SideMenuFilterViewController.swift:97]: This block will never be called. the boolean can only be true or false.

File: BlackNectar/UserPreferences.swift [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+
+class UserPreferences {
+    
+    static let instance = UserPreferences()
+    private init() {}
+    
+    let defaultPreferences = UserDefaults.standard
+    
+    var distanceFilter: Double {
```
Comment (SirWellington) [inline BlackNectar/UserPreferences.swift:19]: Much better.

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 7]
```diff
@@ -380,3 +382,18 @@ fileprivate extension StoresTableViewController {
     
 }
 
+//MARK - Sets User Preferences for various filters
+fileprivate extension StoresTableViewController {
+    
+    func passDefaultValues() {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift]: This is more like `loadDefaultValues()`

File: BlackNectar/UserPreferences.swift [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+        
+        set (value) {
+            
+            defaultPreferences.set(value, forKey: "isOpenNow")
+            
+        }
+        
+    }
+
+    func setSideMenuDefaults(viewController: SideMenuFilterViewController, distanceFilter: Double, isRestaurant: Bool, isOpenNow: Bool, isStore: Bool) {
```
Comment (SirWellington) [inline BlackNectar/UserPreferences.swift]: Don't think this whole function needs to be here.

**Review by SirWellington — COMMENTED** [review]
Great improvement with the computed properties.

File: BlackNectar/BlackNectar/SideMenuFilterViewController.swift (hunk 3 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/SideMenuFilterViewController.swift]: this is fixed

File: BlackNectar/BlackNectar/SideMenuFilterViewController.swift (hunk 5 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/SideMenuFilterViewController.swift:97]: +modified if statements in methods
+cleaned up code

File: BlackNectar/BlackNectar/SideMenuFilterViewController.swift (hunk 4 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/SideMenuFilterViewController.swift:91]: +cleaned up these if statements

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 7 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/StoresTableViewController.swift]: changed name to func loadDefaultValues()

File: BlackNectar/UserPreferences.swift (hunk 8 above)
Comment (cordhernandez) [reply BlackNectar/UserPreferences.swift]: I checked this method it seems that it is needed

File: BlackNectar/BlackNectar/SideMenuFilterViewController.swift (hunk 2 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/SideMenuFilterViewController.swift:58]: + I looked into this, it is a bit confusing but to my knowledge the code is necessary
+ When I removed the code the distance filter and the StoresTableViewController does not work properly

**Pull request merged**